sdks/typescript/src/v1/examples/infered-typing/workflow.ts

```diff
@@ -0,0 +1,50 @@
+import { hatchet } from '../client';
+
+type SimpleInput = {
+  Message: string;
+};
+
+type SimpleOutput = {
+  TransformedMessage: string;
+};
+
+export const declaredType = hatchet.task<SimpleInput, SimpleOutput>({
+  name: 'declared-type',
+  fn: (input) => {
+    return {
+      TransformedMessage: input.Message.toLowerCase(),
+    };
+  },
+});
+
+export const inferredType = hatchet.task({
+  name: 'inferred-type',
+  fn: (input: SimpleInput) => {
+    return {
+      TransformedMessage: input.Message.toUpperCase(),
+    };
+  },
+});
+
+export const inferredTypeDurable = hatchet.durableTask({
+  name: 'inferred-type-durable',
+  fn: async (input: SimpleInput, ctx) => {
+    // await ctx.sleepFor('5s');
+
+    return {
+      TransformedMessage: input.Message.toUpperCase(),
+    };
+  },
+});
+
+export const crazyWorkflow = hatchet.workflow({
+  name: 'crazy-workflow',
+});
+
+const step1 = crazyWorkflow.task(declaredType);
+// crazyWorkflow.task(inferredTypeDurable);
+
+crazyWorkflow.task({
+  parents: [step1],
+  ...inferredType.taskDef,
+});
```

Comment:
IMO, we shouldn't advertise this (or the above thing with `.task()` - I think in this case, people _really_ should be using the DAG toolkit